toolkit/tools/imagecustomizerapi/user.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+)
+
+type User struct {
+	Name                string   `json:"Name"`
+	UID                 string   `json:"UID"`
+	PasswordHashed      bool     `json:"PasswordHashed"`
+	Password            string   `json:"Password"`
+	PasswordPath        string   `json:"PasswordPath"`
+	PasswordExpiresDays int64    `json:"PasswordExpiresDays"`
+	SSHPubKeyPaths      []string `json:"SSHPubKeyPaths"`
+	PrimaryGroup        string   `json:"PrimaryGroup"`
+	SecondaryGroups     []string `json:"SecondaryGroups"`
+	StartupCommand      string   `json:"StartupCommand"`
+}
+
+func (u *User) IsValid() error {
```

Comment:
The toolkit doesn't allow the usage of "mariner" as a user. We should probably align to that here too.

Reply:
Really? I can't find any logic that enforces this requirement.

Reply:
It may be related to https://github.com/microsoft/CBL-Mariner/blob/ff133d6c92b57b88edeb2789f47f3daefceb0f89/toolkit/imageconfigs/additionalconfigs/cloud-init.cfg#L79 but I guess it is fair to not account for that since it is not a toolkit default user.